pyth-sdk-solana/src/state.rs

```diff
@@ -336,18 +339,102 @@ pub struct PriceAccount {
+    pub comp:           [PriceComp; N],
+    /// additional extended account data
+    pub extended:       T,
+}
+
+impl<const N: usize, T> Default for GenericPriceAccount<N, T>
+where
+    T: Default,
+    T: Copy,
+{
+    fn default() -> Self {
+        Self {
+            magic:          Default::default(),
+            ver:            Default::default(),
+            atype:          Default::default(),
+            size:           Default::default(),
+            ptype:          Default::default(),
+            expo:           Default::default(),
+            num:            Default::default(),
+            num_qt:         Default::default(),
+            last_slot:      Default::default(),
+            valid_slot:     Default::default(),
+            ema_price:      Default::default(),
+            ema_conf:       Default::default(),
+            timestamp:      Default::default(),
+            min_pub:        Default::default(),
+            drv2:           Default::default(),
+            drv3:           Default::default(),
+            drv4:           Default::default(),
+            prod:           Default::default(),
+            next:           Default::default(),
+            prev_slot:      Default::default(),
+            prev_price:     Default::default(),
+            prev_conf:      Default::default(),
+            prev_timestamp: Default::default(),
+            agg:            Default::default(),
+            comp:           [Default::default(); N],
+            extended:       Default::default(),
+        }
+    }
 }
 
+impl<const N: usize, T> std::ops::Deref for GenericPriceAccount<N, T>
+where
+    T: Default,
+    T: Copy,
+{
+    type Target = T;
+    fn deref(&self) -> &Self::Target {
+        &self.extended
+    }
+}
+
+#[repr(C)]
+#[derive(Copy, Clone, Debug, Default, Pod, Zeroable, PartialEq, Eq)]
+pub struct PriceCumulative {
+    /// Cumulative sum of price * slot_gap
+    pub price:          i128,
+    /// Cumulative sum of conf * slot_gap
+    pub conf:           u128,
+    /// Cumulative number of slots where the price wasn't recently updated (within
+    /// PC_MAX_SEND_LATENCY slots). This field should be used to calculate the downtime
+    /// as a percent of slots between two times `T` and `t` as follows:
+    /// `(T.num_down_slots - t.num_down_slots) / (T.agg_.pub_slot_ - t.agg_.pub_slot_)`
+    pub num_down_slots: u64,
+    /// Padding for alignment
+    pub unused:         u64,
+}
+
+#[derive(Copy, Clone, Debug, Default, PartialEq, Eq)]
+pub struct PriceAccountExt {
+    pub price_cumulative: PriceCumulative,
+}
+
+/// Backwards compatibility.
+pub type PriceAccount = GenericPriceAccount<32, ()>;
+
+/// Solana-specific Pyth account where the old 32-element publishers are present.
+pub type SolanaPriceAccount = GenericPriceAccount<32, ()>;
+
+/// Pythnet-specific Price accountw ith upgraded 64-element publishers and extended fields.
+pub type PythnetPriceAccount = GenericPriceAccount<64, PriceAccountExt>;
```

Comment:
It's actually 128

Reply:
see https://github.com/pyth-network/pyth-client/blob/f3b78939d768c525d29ed5fcebd767dc13e2fce8/program/c/src/oracle/oracle.h#L27

Reply:
and https://github.com/pyth-network/pyth-client/blob/f3b78939d768c525d29ed5fcebd767dc13e2fce8/program/rust/src/accounts/price.rs#L88

Reply:
but the 64 last are ignored by the on-chain code. The aggregation is only performed with the first 64

Reply:
Fixed.